packages/local_auth/local_auth_darwin/darwin/Tests/FLALocalAuthPluginTests.swift

```diff
@@ -15,547 +16,509 @@ import XCTest
+    mockAlertController = MockAlertController()
 
-    let plugin = FLALocalAuthPlugin(
-      contextFactory: StubAuthContextFactory(contexts: [stubAuthContext]),
-      alertFactory: alertFactory, viewProvider: viewProvider)
+    plugin = LocalAuthPlugin(authContext: mockAuthContext, alertController: mockAlertController)
 
     let strings = createAuthStrings()
-    stubAuthContext.evaluateResponse = true
+    mockAuthContext.evaluatePolicyResult = (true, nil)
 
     let expectation = expectation(description: "Result is called")
     plugin.authenticate(
-      with: FLADAuthOptions.make(
-        withBiometricOnly: false,
+      options: AuthOptions(
+        biometricOnly: false,
         sticky: false,
         useErrorDialogs: false),
       strings: strings
-    ) { resultDetails, error in
+    ) { result in
       XCTAssertTrue(Thread.isMainThread)
-      XCTAssertEqual(resultDetails?.result, FLADAuthResult.success)
-      XCTAssertNil(error)
+
+      switch result {
+      case .success(let resultDetails):
+        XCTAssertEqual(resultDetails.result, AuthResult.success)
+        XCTAssertNil(resultDetails.errorMessage)
+      case .failure(let flutterError):
+        XCTFail("Expected success, but got error: \(flutterError)")
+      }
+
       expectation.fulfill()
     }
     self.waitForExpectations(timeout: timeout)
   }
 
   func testFailedAuthWithBiometrics() {
-    let stubAuthContext = StubAuthContext()
-    let alertFactory = StubAlertFactory()
-    let viewProvider = StubViewProvider()
-    let plugin = FLALocalAuthPlugin(
-      contextFactory: StubAuthContextFactory(contexts: [stubAuthContext]),
-      alertFactory: alertFactory, viewProvider: viewProvider)
+    mockAuthContext = MockAuthContext()
+    mockAlertController = MockAlertController()
+    plugin = LocalAuthPlugin(authContext: mockAuthContext, alertController: mockAlertController)
 
     let strings = createAuthStrings()
-    stubAuthContext.expectBiometrics = true
-    stubAuthContext.evaluateError = NSError(
-      domain: "error", code: LAError.authenticationFailed.rawValue)
+    mockAuthContext.expectBiometrics = true
+    mockAuthContext.evaluatePolicyResult = (
+      false, NSError(domain: "error", code: LAError.authenticationFailed.rawValue)
+    )
 
     let expectation = expectation(description: "Result is called")
     plugin.authenticate(
-      with: FLADAuthOptions.make(
-        withBiometricOnly: true,
-        sticky: false,
-        useErrorDialogs: false),
+      options: AuthOptions(biometricOnly: true, sticky: false, useErrorDialogs: false),
       strings: strings
-    ) { resultDetails, error in
+    ) { resultDetails in
       XCTAssertTrue(Thread.isMainThread)
-      // TODO(stuartmorgan): Fix this; this was the pre-Pigeon-migration
-      // behavior, so is preserved as part of the migration, but a failed
-      // authentication should return failure, not an error that results in a
-      // PlatformException.
```

Comment:
Why was this TODO removed?

Reply:
I removed it while adjusting part of the return in the tests. I put it back on.